### PR TITLE
Use batchgroup, with fixed size

### PR DESCRIPTION
**Pull request by axw**
Based on https://github.com/elastic/apm-aggregation/pull/65, with some modifications:

- Reverted to default MemTableSize (maybe come back to this in a separate change)
- Create batches up front, which are reset and reused on commit
- Maintain three lists of batches: empty, full, and active (non-empty, non-full). We also maintain a counting semaphore to coordinate waiting for availability

### CODE REVIEW

**Comment by axw** [comment]
```
goos: linux
goarch: amd64
pkg: github.com/elastic/apm-aggregation/aggregators
cpu: AMD Ryzen 7 PRO 5850U with Radeon Graphics     
                                  │ /tmp/old.txt │            /tmp/new.txt            │
                                  │    sec/op    │   sec/op     vs base               │
NDJSONParallel/go-2.0.0.ndjson       14.21m ± 5%   13.73m ± 6%        ~ (p=0.394 n=6)
NDJSONParallel/go-2.0.0.ndjson-16    9.934m ± 3%   4.748m ± 4%  -52.21% (p=0.002 n=6)
geomean                              11.88m        8.074m       -32.03%

                                  │ /tmp/old.txt  │            /tmp/new.txt            │
                                  │     B/op      │     B/op      vs base              │
NDJSONParallel/go-2.0.0.ndjson      7.035Mi ±  1%   6.972Mi ± 1%       ~ (p=0.058 n=6)
NDJSONParallel/go-2.0.0.ndjson-16   6.250Mi ± 21%   7.025Mi ± 2%       ~ (p=0.065 n=6)
geomean                             6.631Mi         6.998Mi       +5.54%

                                  │ /tmp/old.txt │            /tmp/new.txt             │
                                  │  allocs/op   │  allocs/op   vs base                │
NDJSONParallel/go-2.0.0.ndjson       8.000 ±  0%   8.000 ±  0%       ~ (p=1.000 n=6) ¹
NDJSONParallel/go-2.0.0.ndjson-16   10.000 ± 10%   9.000 ± 56%       ~ (p=0.080 n=6)
geomean                              8.944         8.485        -5.13%
¹ all samples are equal
```

(old.txt is https://github.com/elastic/apm-aggregation/pull/65, new.txt is this branch)

File: aggregators/aggregator_test.go [hunk 1]
```diff
@@ -1145,7 +1145,10 @@ func BenchmarkAggregateBatchSerial(b *testing.B) {
 func BenchmarkAggregateBatchParallel(b *testing.B) {
 	b.ReportAllocs()
 	agg := newTestAggregator(b)
-	defer agg.Close(context.Background())
+	go func() {
+		agg.Run(context.Background())
+	}()
+	// defer agg.Close(context.Background())
```
Comment (lahsivjar) [inline aggregators/aggregator_test.go:1151]: Just a heads up, the main benchmark should also be run with this change or both should be run without this change. I found that using `agg.Close` was pushing all the complexity of flushing the multiple batches we create in the benchmark result. My thinking for doing this change was that the `Close` will not be in hot path.

**Review by lahsivjar — APPROVED** [review]
Love the approach!!

**Comment by axw** [comment]
This increases memory usage rather a lot, for fairly minimal throughput benefit. Closing, may come back to parallelism later.

**Pull request closed without merging**